src/app/clusters/diagnostic-logs-server/Upgrading.md

```diff
@@ -0,0 +1,41 @@
+### DiagnosticLogsCluster API Changes
+
+The Diagnostic Logs cluster is a device-wide singleton. There is no need to
+manage delegates on a per-endpoint basis. To reflect this, the API has been
+simplified.
+
+What Changed
+
+#### SetDelegate() API Updated
+
+Old API (deprecated):
+
+```
+chip::app::Clusters::DiagnosticLogs::DiagnosticLogsServer::SetDiagnosticLogsProviderDelegate(endpointId, delegate);
+```
+
+-   Required passing an EndpointId.
+-   Misleading, since Diagnostic Logs is not endpoint-scoped.
+
+New API:
+
+```
+chip::app::Clusters::DiagnosticLogsCluster::Instance().SetDelegate(delegate);
```

Comment:
There is no Instance() call though.

If we use CodegenIntegration we may want to provide an instance. otherwise it is up to the app.